task02/test_runner.h

```diff
@@ -0,0 +1,46 @@
+#include <sstream>
+#include <exception>
+#include <iostream>
+#include <string>
+#include <map>
+#include <set>
+
+template<class T, class U>
+void Assert(const T& t, const U& u,
+    const std::string& hint)
+{
+  if (t != u) {
+    std::ostringstream os;
+    os << "Assertion failed: " << t << " != " << u
+       << " hint: " << hint;
+    throw std::runtime_error(os.str());
+  }
+}
+
+class TestRunner {
+public:
+  template <class TestFunc>
```

Comment:
Good test class!
But here we don't need a template at all. 
Maybe, if you pass some parameters to TestFunc, it will be a case.

Reply:
I just don't know how to get function as an argument for function. That' the only reason why I've used template.

Reply:
In such case you can try this:
` void RunTest(void (*func)(), const std::string& test_name)`

This strongly declares function arguments.